### PR TITLE
update adjustPacketRateForBaud for new rf modes

### PR DESCRIPTION
**Pull request by JyeSmith**
This updates adjustPacketRateForBaud() to be based on the handset interval and not specific rf modes.  A condition for 400k baud and 1kHz rf mode has also been added.

### CODE REVIEW

**Review by pkendall64 — APPROVED** [review]
Looks good to me!

File: src/src/tx_main.cpp [hunk 1]
```diff
@@ -404,7 +404,20 @@ uint8_t adjustPacketRateForBaud(uint8_t rateIndex)
...
         expresslrs_mod_settings_s const * const ModParams = get_elrs_airRateConfig(rateIndex);
-        if (ModParams->enum_rate <= RATE_LORA_250HZ) {
+        uint32_t RCpacketInterval = ModParams->interval * ModParams->numOfSends;
+        if (RCpacketInterval >= 4000) {
+          break;
+        }
+        rateIndex++;
+      }
+    }
+    else
```
Comment (schugabe) [inline src/src/tx_main.cpp:414]: I'd add { } to the block otherwise it's hard to read

Comment (JyeSmith) [reply src/src/tx_main.cpp]: Hows it look now?

Im also happy if you would like to make the changes directly if Ive misunderstood.

**Comment by CapnBry** [comment]
This is a great idea to both fix F1000 being too fast for 400k as well as making it based on something smarter, the expected handset interval!

The code is too copy paste though right? So Instead of having this
```
      while (rateIndex < RATE_MAX)
      {
        expresslrs_mod_settings_s const * const ModParams = get_elrs_airRateConfig(rateIndex);
        uint32_t RCpacketInterval = ModParams->interval * ModParams->numOfSends;
        if (RCpacketInterval >= 4000)
        {
          break;
        }
        rateIndex++;
      }
```
and this **identical code** except for the threshold
```
      while (rateIndex < RATE_MAX)
      {
        expresslrs_mod_settings_s const * const ModParams = get_elrs_airRateConfig(rateIndex);
        uint32_t RCpacketInterval = ModParams->interval * ModParams->numOfSends;
        if (RCpacketInterval >= 2000)
        {
          break;
        }
        rateIndex++;
      }
````

Just make a new function in common.cpp
```
uint8_t get_elrs_HandsetRate_max(uint8_t rateIndex, uint32_t minInterval)
{
    while (rateIndex < RATE_MAX)
    {
        expresslrs_mod_settings_s const * const ModParams = &ExpressLRS_AirRateConfig[rateIndex];
        // Handset interval = time between packets from handset, which is expected to be air rate * number of times it is sent
        uint32_t handsetInterval = ModParams->interval * ModParams->numOfSends;
        if (handsetInterval <= minInterval)
            break;
        ++rateIndex;
    }

    return rateIndex;
}
```

so the code just becomes
```
if (crsf.GetCurrentBaudRate() == 115200) // Packet rate limited to 250Hz if we are on 115k baud
  return get_elrs_HandsetRate_max(rateIndex, 4000);
// etc
```

This works out to be less code, is easier to since each is just 2 lines, and is also slightly faster due to the lack of having to go out to `get_elrs_airRateConfig` multiple times (not that that matters at all, but hey it is faster and free, which is everything we're about).

**Comment by JyeSmith** [comment]
@CapnBry If you have a patch done and tested Im fine for you to push this approach.

**Comment by CapnBry** [comment]
> @CapnBry If you have a patch done and tested Im fine for you to push this approach.

Can do, and did! Didn't want to push it if you didn't think it was apropos.

**Review by pkendall64 — APPROVED** [review]
Even more better!

**Review by CapnBry — APPROVED** [review]
Does the same thing that we currently do for the 115200 >250Hz. 

* Tested on OpenTX 2.3.14 Taranis Q X7 at 400k, F1000 drops to D500, all other rates work.
* Tested on OpenTX 2.3.14 Taranis Q X7 at 115k, F1000, F500, D500 drops to D250, all other rates work.
* Tested on Radiomaster Zorro 921k, all rates work fine.

There's a problem that the user can set a rate that they can do, then lower the baud and therefore lock themselves out of the module, but that's existing. We can always come back at a later date and fix that but I don't think it is necessary.

**Comment by JyeSmith** [comment]
Hmm good thinking, but I can see Deadbyte pulling he hair out with this support request.  Should we just add another call after autobaud() to catch this?

https://github.com/ExpressLRS/ExpressLRS/blob/d3081ca7a0330c9d70a263dee4be613ff6413290/src/lib/CRSF/CRSF.cpp#L847

**Comment by CapnBry** [comment]
> Should we just add another call after autobaud() to catch this?

It shouldn't have to. The UARTconnected handler is supposed to fix this because it calls SetRFLinkRate() but I am not sure why that's not working. It theoretically should update the packet rate when CRSF packets start coming in at a new baud (rather than possibly changing rates every time it cycles a baud).

Actually, I'll take a look at that real quick.

**Comment by CapnBry** [comment]
It actually is working, it is just that the OLED display always shows the configured air rate and not the actual air rate. The Lua and Display apparently operate differently:
* The Lua will degrade the air rate and save that to config, and will also pretend that rate is the configured rate (shows it as the configured rate)
* The Display will set whatever you tell it to and save it, and will only show the configured rate no matter what.

This should be brought to parity at some point so the behavior is the same between the two but outside the scope of this PR. I also just noticed that some dummy put "F500" instead of "F500Hz" in the display (ashamed).

But to clarify, it seems to properly degrade the connection on startup, the display just doesn't show it so I got confused since the Lua does. Therefore this PR is good to go.

**Review by wvarty — APPROVED** [review]

**Pull request merged**